spark/src/main/java/org/apache/iceberg/actions/BaseSparkAction.java

```diff
@@ -86,6 +126,32 @@
     return manifestDF.union(otherMetadataFileDF).union(manifestListDF);
   }
 
+  protected static Dataset<Row> loadMetadataTable(SparkSession spark, String tableName, String tableLocation,
+                                                  MetadataTableType type) {
+    DataFrameReader noCatalogReader = spark.read().format("iceberg");
+    if (tableName.contains("/")) {
+      // Metadata location passed, load without a catalog
+      return noCatalogReader.load(tableName + "#" + type);
+    } else {
+      // Try catalog based name based resolution
+      try {
+        return spark.table(tableName + "." + type);
+      } catch (Exception e) {
+        // Catalog based resolution failed, our catalog may be a non-DatasourceV2 Catalog
+        if (tableName.startsWith("hadoop.")) {
```

Comment:
This logic is just for Spark 2.x right? If so then maybe we can add a comment explaining it and noting when we can remove it.

Reply:
Anton I were talking about this, the issue is that it isn't just for SparkV2 as it's quite possible for a user to be using manual catalogs in a Spark3 application (or ported Spark2 application.) If we remove this we break anyone who is accessing tables by this method. 

I think that's actually something that's fine to break eventually but I think it would be pretty big breaking change.

Example

Actions.forTable(tableCreatedFromHadoopOrHiveCatalog)

Reply:
Sounds reasonable to me.